### PR TITLE
AEA-1115 MAM overrides support

### LINKED ISSUE #1975: Improve AgentConfig.update method
**Is your feature request related to a problem? Please describe.**
AgentConfig.update method does not support all the fields

**Describe the solution you'd like**
make update method compatible with all fields supported in config

**Additional context**
MultiAgentManager uses it, but now implemented with workaround and dict patching instead of AgentConfig.update

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

MAM: better configuration handling + refactoring + build command support

## Fixes
#1975 #1986 #2080

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/aea_builder.py [hunk 1]
```diff
@@ -1570,45 +1573,20 @@ def from_aea_project(
...
 
-        builder.set_from_configuration(
-            agent_configuration, aea_project_path, skip_consistency_check
+        verify_or_create_private_keys(
+            aea_project_path=aea_project_path, exit_on_error=False
         )
-        return builder
-
-    @classmethod
-    def from_config_json(
```
Comment (DavidMinarsch) [inline aea/aea_builder.py:1591]: Why removed?

File: aea/cli/config.py [hunk 2]
```diff
@@ -342,3 +343,23 @@ def convert_value_str_to_type(value: str, type_str: str) -> JSON_TYPES:
...
+            component_config = self.load_component_configuration(
+                component_id, skip_consistency_check=True
+            )
+            obj.update(component_config.get_overridable())
+            obj.update(
+                deepcopy(
+                    self.agent_config.component_configurations.get(component_id, {})
+                )
+            )
+            if obj:
```
Comment (DavidMinarsch) [inline aea/cli/config.py:363]: How can it not be true?

File: aea/manager/__init__.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the manager modules."""
+from .manager import MultiAgentManager
```
Comment (DavidMinarsch) [inline aea/manager/__init__.py:20]: full path!

File: aea/manager/project.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+        """Get builder instance."""
+        builder = self._get_builder(self.agent_config, self.project.path)
+        if not builder.private_key_paths:
+            # no keys, but create one for builder only
+            default_ledger = builder.get_default_ledger()
+            builder.add_private_key(
+                default_ledger, self._create_private_key(default_ledger)
+            )
+        builder.set_name(self.agent_name)
+        builder.set_runtime_mode("threaded")
```
Comment (DavidMinarsch) [inline aea/manager/project.py:160]: constantize

File: aea/manager/project.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+                {"version": "any", **component_override}
+            )
+            component_override.pop("author")
+            component_override.pop("name")
+            component_override.pop("type")
+            component_override.pop("version")
+            component_configurations[component_id] = component_override
+
+        overrides["component_configurations"] = component_configurations
+        return self.agent_config_manager.update_config(overrides)
```
Comment (DavidMinarsch) [inline aea/manager/project.py:226]: this could fail..provide proper error message

File: tests/test_aea_builder.py [hunk 6]
```diff
@@ -668,13 +667,6 @@ def test_from_project(self):
         dummy_model = dummy_skill.models["dummy"]
         assert dummy_model.config == {"model_arg_1": 42, "model_arg_2": "2"}
 
-    def test_from_json(self):
-        """Test load project from json file with path specified."""
-        with open(Path(self._get_cwd(), DEFAULT_AEA_CONFIG_FILE), "r") as fp:
-            json_config = yaml_load_all(fp)
-
-        AEABuilder.from_config_json(json_config, Path(self._get_cwd()))
-
```
Comment (DavidMinarsch) [inline tests/test_aea_builder.py:677]: I think we should keep this functionality. It is a good functionality to have.

File: tests/test_manager.py [hunk 7]
```diff
@@ -33,14 +34,17 @@
 from tests.conftest import MY_FIRST_AEA_PUBLIC_ID
 
 
-class TestMultiAgentManagerAsyncMode:  # pylint: disable=unused-argument,protected-access,attribute-defined-outside-init
+@patch("aea.aea_builder.AEABuilder.install_pypi_dependencies")
+class TestMultiAgentManagerAsyncMode(
+    TestCase
```
Comment (DavidMinarsch) [inline tests/test_manager.py:39]: What's the benefit of also using TestCase now...?

**Review by DavidMinarsch — COMMENTED** [review]
LGTM, some open comments

File: aea/aea_builder.py (hunk 1 above)
Comment (solarw) [reply aea/aea_builder.py:1591]: not used at all

File: aea/cli/config.py (hunk 2 above)
Comment (solarw) [reply aea/cli/config.py:363]: protocol has no overridables so set of overridables can be empty, checked  during tests

File: aea/manager/project.py (hunk 4 above)
Comment (solarw) [reply aea/manager/project.py:160]: this mode is not an option, this is one and the only supported and not supposed to be changed

File: tests/test_aea_builder.py (hunk 6 above)
Comment (solarw) [reply tests/test_aea_builder.py:677]: it was used only by my code i replaced

File: tests/test_manager.py (hunk 7 above)
Comment (solarw) [reply tests/test_manager.py:39]: got some issues with teardown/setup so decided to use TestCase as a bse class, actually it's a proper test case definition.

File: aea/manager/project.py (hunk 5 above)
Comment (solarw) [reply aea/manager/project.py]: done, checks for Value and Key errors

File: aea/manager/__init__.py (hunk 3 above)
Comment (solarw) [reply aea/manager/__init__.py]: done

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by panasevychol — APPROVED** [review]
#2080 
Other is good. Tested with GUI app, works well.

**Pull request merged**